go/cmd/dolt/commands/sqlserver/sqlserver.go

```diff
@@ -235,10 +235,33 @@ func startServer(ctx context.Context, versionStr, commandStr string, args []stri
 // GetServerConfig returns ServerConfig that is set either from yaml file if given, if not it is set with values defined
 // on command line. Server config variables not defined are set to default values.
 func GetServerConfig(dEnv *env.DoltEnv, apr *argparser.ArgParseResults) (ServerConfig, error) {
+	var yamlCfg YAMLConfig
 	if cfgFile, ok := apr.GetValue(configFileFlag); ok {
-		return getYAMLServerConfig(dEnv.FS, cfgFile)
+		cfg, err := getYAMLServerConfig(dEnv.FS, cfgFile)
+		if err != nil {
+			return nil, err
+		}
+		yamlCfg = cfg.(YAMLConfig)
+	} else {
+		return getCommandLineServerConfig(dEnv, apr)
+	}
+
+	// a user/pass word was specified in yamlCfg
+	if yamlCfg.UserConfig.Name != nil || yamlCfg.UserConfig.Password != nil {
```

Comment:
Isn't this backwards? We want the command line args to override whatever's in the yaml config